app/src/main/java/de/danoeh/antennapod/preferences/PreferenceUpgrader.java

```diff
@@ -68,6 +69,12 @@ private static void upgrade(int oldVersion) {
                     break;
             }
         }
+        if (oldVersion < 1070397) {
```

Comment:
Please add this to the 1.7.4 block, so all updates for one version are in one location.

Reply:
If I change it to 1.7.4 block, without changing the `versionCode` in `app/build.gradle` to 1.7.4, when someone runs the app with the code,
- the upgrade logic will run repeatedly (as the version code is still in 1.7.3 range). If an user changes the settings, it will be overwritten by the upgrade logic when the app launches again.
- Actually, the same issue applies to the current 1.7.4 block. If an user sets the queue lock to `true`, the upgrade logic will repeatedly change it back to `false`
```java
        if (oldVersion < 1070400) {
            int theme = UserPreferences.getTheme();
            if (theme == R.style.Theme_AntennaPod_Light) {
                prefs.edit().putString(UserPreferences.PREF_THEME, "system").apply();
            }

            UserPreferences.setQueueLocked(false);
        }
```

Reply:
That whole version code thing is not easy anyways because we either need to anticipate the new version number (currently we add with 1.7.4 but I think the new release is worth 1.8.0) or we bump to something that is officially a RC versionCode (looking at the schema in build.gradle)

Reply:
Okay, so you're saying that we punt the issue for now, to avoid "polluting" versionCode.

---

What about a revised scheme going forward (starting for next release)?

Option 1: a tweak of  current scheme, allowing rooms for version bump in `develop`
```
   "1.2.3-RC1"      -> 1020301
   "1.2.3-RC2"      -> 1020302   (up to RC9)
   "1.2.3"          -> 1020310
develop, post 1.2.3 : no versionCode change, but change versionName to indicate it.
   "1.2.3-SNAPSHOT" -> 1020310
develop post 1.2.3, preference upgrade is needed
   "1.2.3-SNAPSHOT" -> 1020311    (bump the last digit, up to 9 such bump is okay)
in case a minor fix of the release is needed
 "1.2.3b"          -> 1020320     (hotfix release. For actual releases, the last digit is always 0)
```

Option 2: decouple versionCode from versionName.

Reply:
In the future, I want to make public releases 1.8.0/1.9.0 and bugfixes 1.8.1/1.8.2 instead of appending a letter (a bit like semantic versioning). If you look at the releases page, we wanted to remove the fourth variable from version names a few releases back but failed to do so (replaced with letters).

Not totally sure how we should handle RC versions there.

Reply:
I'm not sure what the releases page is. The only release-related page I know is https://github.com/AntennaPod/AntennaPod/wiki/Releasing-a-New-Version .

Maybe it's a page I don't have access to?

Reply:
I meant this page: https://github.com/AntennaPod/AntennaPod/releases. This page shows the version name changes even better: https://github.com/AntennaPod/AntennaPod/tags